### ISSUE TITLE
Expand config utilities and test

### ISSUE THREAD

**Issue by lockefox** [EVEprosper/ProsperCommon#2]
Config utilities need to be expanded to support logger's desired option priority needs

Expected priority order:

1. (non default) function args
2. script-local version of .cfg object
3. global/git version of .cfg object

Need to add a class for easy option parsing.  And tests to validate for release

**Comment by lockefox** [comment]
scrap/prototype 

```python
class ConfigValuePicker(object):
    '''helper class to pick between local/global config values'''
    def __init__(
            self,
            global_config,
            local_config=None,
            logger=DEFAULT_LOGGER
    ):
        self.logger = logger
        self.local_config = local_config
        self.global_config = global_config

        self.global_config_cache = {}
        self.local_config_cache = {}

    def load_local_config(
            self,
            local_config
    ):
        '''at ConfigValuePicker setup time, load/override a local config'''
        if self.local_config:
            self.logger.debug('overriding existing local_config')
        self.local_config = local_config
        self.local_config_cache = {}

    def cachebuster(self):
        '''clear both cached lists'''
        self.global_config_cache = {}
        self.local_config_cache = {}

    def cachebuster_global(self):
        '''clear global cache'''
        self.global_config_cache = {}

    def cachebuster_local(self):
        '''clear local cache'''
        self.local_config_cache = {}

    def get_value(
            self,
            config_section,
            config_key,
            local_argument=None,
            local_argument_default=None
    ):
        '''parse configs for preferred value
        PRIORITY ORDER:
            1. local_argument (ideally passed from requestor's function args)
            2. local_config (passed from requestor's config object)
            3. global_config (loaded at init time as defined)

        RETURN: value as per priority order
        '''
        if local_argument != local_argument_default:
            self.logger.debug('local_argument is non-default, returning local_argument')
            return local_argument   #1. local argument (non default)

        local_option = self.check_cache(
            self.local_config_cache,
            config_section
            config_key
        )
        if local_option is not None:
            return local_option
        else:
            try:
                local_option = self.local_config[config_section][config_key]
            except Exception as error_msg:
                warning_msg = \
                    'section/key pair missing in local config' + \
                    '\r\texception={0}'.format(error_msg) + \
                    '\r\tconfig_key={0}.{1}'.format(config_section, config_key)
                self.logger.warning(warning_msg)
                #no warnings.warn, okay to be missing from local config
        if local_option is not None:
            self.update_cache(
                self.local_config_cache,
                config_section,
                config_key
            )
            return local_option

        global_option = None
        try:
            global_option = self.global_config[config_section][config_key]
        except Exception as error_msg:
            warning_msg = \
                'section/key pair missing in global config' + \
                '\r\texception={0}'.format(error_msg) + \
                '\r\tconfig_key={0}.{1}'.format(config_section, config_key)
            self.logger.warning(warning_msg)
            warnings.warn(warning_msg, RuntimeWarning)
            #all requested keys should be in GLOBAL config
        if global_option:
            return global_option

        # If all else fails, return local_argument_default
        return local_argument_default
```

**Comment by lockefox** [comment]
Feature added/test/released for v0.3.0

3a0a3d8b6e374998c8bc44f6c503cdf952377921
